From a3fc9bdbb28f1fd49faab43418a755541c14d969 Mon Sep 17 00:00:00 2001
From: Yazan Shhady <yazan.shhady@solid-run.com>
Date: Wed, 9 Aug 2023 11:35:45 +0300
Subject: [PATCH 1/3] rzg2l-humminboard-ripple: update dts

- fix BT
- enable WiFi 32MHz Clock
- splite dts nodes between SOM & Hummingboard-Ripple
---
 .../dts/renesas/rzg2l-hummingboard-ripple.dts |  72 +++++++++---
 .../dts/renesas/rzg2l-sr-pinfunction.dtsi     |  14 +--
 arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi | 111 ++++++++----------
 4 files changed, 125 insertions(+), 91 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/rzg2l-hummingboard-ripple.dts b/arch/arm64/boot/dts/renesas/rzg2l-hummingboard-ripple.dts
index f7afc39fa8a1..fdecb0caaa83 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-hummingboard-ripple.dts
+++ b/arch/arm64/boot/dts/renesas/rzg2l-hummingboard-ripple.dts
@@ -21,23 +21,67 @@
 #include "rzg2l-sr-pinfunction.dtsi"
 
 / {
-  model = "Solidrun RZ/G2L HummingBaord-Ripple";
-  compatible = "renesas,r9a07g044l2", "renesas,r9a07g044";
+	model = "Solidrun RZ/G2L HummingBaord-Ripple";
+	compatible = "renesas,r9a07g044l2", "renesas,r9a07g044";
 
-  x1_clk: x1-clock {
+	x1_clk: x1-clock {
 		compatible = "fixed-clock";
 		#clock-cells = <0>;
 		clock-frequency = <24000000>;
 	};
+
+	hdmi-out {
+		compatible = "hdmi-connector";
+		type = "d";
+
+		port {
+			hdmi_con_out: endpoint {
+			remote-endpoint = <&adv7535_out>;
+			};
+		};
+	};
 };
 
 &pinctrl {
-  gpio-lte_on {
-    gpio-hog;
-    gpios = <RZG2L_GPIO(19, 1) GPIO_ACTIVE_HIGH>;
-    output-high;
-    line-name = "gpio_lte_on";
-  };
+	gpio-sd0-pwr-en-hog {
+		gpio-hog;
+		gpios = <RZG2L_GPIO(4, 1) GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "gpio_sd0_pwr_en";
+	};
+
+	gpio-lte_on {
+		gpio-hog;
+		gpios = <RZG2L_GPIO(17, 0) GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "gpio_lte_on";
+	};
+
+	gpio-lte_reset {
+		gpio-hog;
+		gpios = <RZG2L_GPIO(42, 2) GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "gpio_lte_reset";
+	};
+
+	/*
+	 * The below switch logic can be used to select the device between
+	 * eMMC and microSD, after setting GPIO_SD0_DEV_SEL to high in DT.
+	 * HB-Ripple: S3[1] should be at OFF position to enable eMMC
+	 * HB-Ripple: S3[1] should be at position ON to enable uSD card
+	 */
+	
+	/*
+	 * TODO: implement device-selection for SolidRun SoM / Carrier
+	 *
+	 * gpio-sd0-dev-sel-hog {
+	 * 	gpio-hog;
+	 * 	gpios = <RZG2L_GPIO(22, 1) GPIO_ACTIVE_HIGH>;
+	 * 	output-high;
+	 * 	line-name = "gpio_sd0_dev_sel";
+	 * };
+	 */
+
 };
 
 &i2c0 {
@@ -92,11 +136,11 @@ rtc@69 {
 		abracon,tc-resistor = <3>;
 	};
 
-  eeprom_carrier: eeprom@57 {
-    compatible = "st,24c02", "atmel,24c02";
-    reg = <0x57>;
-    pagesize = <16>;
-  };
+	eeprom_carrier: eeprom@57 {
+		compatible = "st,24c02", "atmel,24c02";
+		reg = <0x57>;
+		pagesize = <16>;
+	};
 
 };
 
diff --git a/arch/arm64/boot/dts/renesas/rzg2l-sr-pinfunction.dtsi b/arch/arm64/boot/dts/renesas/rzg2l-sr-pinfunction.dtsi
index a9faa77d8203..b202f08229c0 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-sr-pinfunction.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-sr-pinfunction.dtsi
@@ -46,13 +46,6 @@ scif2_pins: scif2 {
 			 <RZG2L_PORT_PINMUX(48, 4, 1)>; /* RTS# */
 	};
 
-	sd1-pwr-en-hog {
-		gpio-hog;
-		gpios = <RZG2L_GPIO(39, 2) GPIO_ACTIVE_HIGH>;
-		output-high;
-		line-name = "sd1_pwr_en";
-	};
-
 	sdhi1_pins: sd1 {
 		sd1_data {
 			pins = "SD1_DATA0", "SD1_DATA1", "SD1_DATA2", "SD1_DATA3";
@@ -112,14 +105,11 @@ ssi1_pins: ssi1 {
 	};
 
 	usb0_pins: usb0 {
-		pinmux = <RZG2L_PORT_PINMUX(4, 0, 1)>, /* VBUS */
-			 <RZG2L_PORT_PINMUX(5, 0, 1)>, /* OVC */
-			 <RZG2L_PORT_PINMUX(5, 1, 1)>; /* OTG_ID */
+		pinmux = <RZG2L_PORT_PINMUX(4, 0, 1)>; /* VBUS */
 	};
 
 	usb1_pins: usb1 {
-		pinmux = <RZG2L_PORT_PINMUX(42, 0, 1)>, /* VBUS */
-			 <RZG2L_PORT_PINMUX(42, 1, 1)>; /* OVC */
+		pinmux = <RZG2L_PORT_PINMUX(42, 0, 1)>;/* VBUS */
 	};
 
 };
diff --git a/arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi b/arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi
index 9160f38c23ff..4aa0973f852c 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-sr-som.dtsi
@@ -97,13 +97,12 @@ reg_1p1v: regulator-vdd-core {
 
 	vccq_sdhi0: regulator-vccq-sdhi0 {
 		compatible = "regulator-gpio";
-
 		regulator-name = "SDHI0 VccQ";
 		regulator-min-microvolt = <1800000>;
 		regulator-max-microvolt = <3300000>;
 		states = <3300000 1>, <1800000 0>;
 		regulator-boot-on;
-		//gpios = <&pinctrl RZG2L_GPIO(39, 0) GPIO_ACTIVE_HIGH>;
+		gpios = <&pinctrl RZG2L_GPIO(39, 0) GPIO_ACTIVE_HIGH>;
 		regulator-always-on;
 	};
 
@@ -134,17 +133,6 @@ vccq_sdhi1: regulator-vccq-sdhi1 {
     states = <3300000 1>, <1800000 0>;
   };
 
-  hdmi-out {
-    compatible = "hdmi-connector";
-    type = "d";
-
-    port {
-      hdmi_con_out: endpoint {
-        remote-endpoint = <&adv7535_out>;
-      };
-    };
-  };
-
 };
 
 &eth0 {
@@ -153,13 +141,10 @@ &eth0 {
 	phy-handle = <&phy0>;
 	phy-mode = "rgmii-id";
 	status = "okay";
+	renesas,no-ether-link;
 
 	phy0: ethernet-phy@0 {
 		reg = <0>;
-/*
-		interrupt-parent = <&pinctrl>;
-		interrupts = <RZG2L_GPIO(27, 0) IRQ_TYPE_LEVEL_LOW>;
-*/
 	};
 };
 
@@ -169,13 +154,10 @@ &eth1 {
 	phy-handle = <&phy1>;
 	phy-mode = "rgmii-id";
 	status = "okay";
+	renesas,no-ether-link;
 
 	phy1: ethernet-phy@0 {
 		reg = <0>;
-/*
-		interrupt-parent = <&pinctrl>;
-    interrupts = <RZG2L_GPIO(42, 4) IRQ_TYPE_LEVEL_LOW>;
-*/
 	};
 };
 
@@ -232,13 +214,6 @@ eth1_pins: eth1 {
        <RZG2L_PORT_PINMUX(36, 0, 1)>; /* ET1_RXD3 */
   };
 
-	gpio-sd0-pwr-en-hog {
-		gpio-hog;
-		gpios = <RZG2L_GPIO(4, 1) GPIO_ACTIVE_LOW>;
-		output-high;
-		line-name = "gpio_sd0_pwr_en";
-	};
-
 	qspi0_pins: qspi0 {
 		qspi0-data {
 			pins = "QSPI0_IO0", "QSPI0_IO1", "QSPI0_IO2", "QSPI0_IO3";
@@ -252,25 +227,38 @@ qspi0-ctrl {
 	};
 
 	/*
-	 * The below switch logic can be used to select the device between
+	 * The switch logic can be used to select the device between
 	 * eMMC and microSD, after setting GPIO_SD0_DEV_SEL to high in DT.
-	 * S3[1] should be at OFF position to enable eMMC
-	 * S3[1] should be at position ON to enable uSD card
 	 */
 
 	/*
 	 * TODO: implement device-selection for SolidRun SoM / Carrier
 	 *
-	 * gpio-sd0-dev-sel-hog {
-	 * 	gpio-hog;
-	 * 	gpios = <RZG2L_GPIO(22, 1) GPIO_ACTIVE_HIGH>;
-	 * 	output-high;
-	 * 	line-name = "gpio_sd0_dev_sel";
-	 * };
 	 */
 
+#if SW_SD0_DEV_SEL
+/* eMMC */	
+	/* P22_1 - SD0_DEV_SEL (High: [eMMC]; Low: [uSD]) */
+	gpio-sd0-dev-sel-hog {
+		gpio-hog;
+		gpios = <RZG2L_GPIO(22, 1) GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "gpio_sd0_dev_sel";
+	};	
+
+	/* P39_0 - LDO_SEL1 (High: 3.3v [uSD]; Low: 1.8v [eMMC]) */
+	gpio-sd0-vdd-18v-hog {
+		gpio-hog;
+		gpios = <RZG2L_GPIO(39, 0) GPIO_ACTIVE_LOW>;
+		output-high;
+		line-name = "gpio_sd0_vdd_1.8v";
+		status = "okay";
+	};
+
+#endif
+
   /* BT_REG_ON*/
-  gpio-sd1-bt_reg_on {
+    gpio-sd1-bt_reg_on {
     gpio-hog;
     gpios = <RZG2L_GPIO(23, 0) GPIO_ACTIVE_HIGH>;
     output-high;
@@ -343,8 +331,10 @@ &sdhi0 {
 
 	vmmc-supply = <&reg_3p3v>;
 	vqmmc-supply = <&vccq_sdhi0>;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
 	bus-width = <4>;
-  max-frequency = <50000000>;
+	max-frequency = <50000000>; /* 50MiB */
 	status = "okay";
 };
 #endif
@@ -373,30 +363,25 @@ &vccq_sdhi1 {
 
 /* WiFi - CYW43439 */
 &sdhi1 {
-  #address-cells = <1>;
-  #size-cells = <0>;
+	#address-cells = <1>;
+	#size-cells = <0>;
 	pinctrl-0 = <&sdhi1_pins>;
 	pinctrl-1 = <&sdhi1_pins_uhs>;
 	pinctrl-names = "default", "state_uhs";
-  status = "okay";
-
-  /*
-  cap-power-off-card;
-  pm-ignore-notify;
-  keep-power-in-suspend;
-  */
-  non-removable;
+	status = "okay";
+
+	non-removable;
 	vmmc-supply = <&reg_3p3v>;
 	vqmmc-supply = <&vccq_sdhi1>;
 	bus-width = <4>;
-  max-frequency = <50000000>;
-  fsl,delay-line = <12>;
-  keep-power-in-suspend;
+	max-frequency = <50000000>;
+	fsl,delay-line = <12>;
+	keep-power-in-suspend;
 
-  brcmf: wifi@1 {
-          reg = <1>;
-          compatible = "brcm,bcm4329-fmac";
-  };
+	brcmf: wifi@1 {
+		reg = <1>;
+		compatible = "brcm,bcm4329-fmac";
+	};
 };
 
 &wdt0 {
@@ -478,19 +463,17 @@ &i2c3 {
 	pmic: raa215300@12 {
 		compatible = "renesas,raa215300";
 		reg = <0x12>;
-
 		rtc-enable;
+		mpio2-32k-enable;
 	};
 
 	rtc: isl1208@6f {
 		compatible = "isil,isl1208";
 		reg = <0x6f>;
-
 		//external-oscillator;
+		status="disabled";
 	};
 
-
-
 };
 
 &ohci0 {
@@ -526,9 +509,8 @@ &scif2 {
 
   bluetooth {
     pinctrl-names = "default";
-    compatible = "brcm,bcm43439-bt";
-    //host-wakeup-gpios = <&gpio1 28 0>;
-    max-speed = <4000000>;
+	max-speed = <115200>;
+	compatible = "brcm,bcm4330-bt";
   };
 };
 
@@ -544,7 +526,6 @@ &spi1 {
 &usb2_phy0 {
 	pinctrl-0 = <&usb0_pins>;
 	pinctrl-names = "default";
-
 	vbus-supply = <&usb0_vbus_otg>;
 	status = "okay";
 };
@@ -552,7 +533,7 @@ &usb2_phy0 {
 &usb2_phy1 {
 	pinctrl-0 = <&usb1_pins>;
 	pinctrl-names = "default";
-  vbus-supply = <&usb1_vbus>;
+  	vbus-supply = <&usb1_vbus>;
 	status = "okay";
 };

-- 
2.25.1

